src/targets/gpu/kernels/include/migraphx/kernels/index.hpp

```diff
@@ -124,11 +113,7 @@ struct index
 #ifdef MIGRAPHX_NLOCAL
     constexpr index_constant<MIGRAPHX_NLOCAL> max_nlocal() const { return {}; }
 #else
-    __device__ index_int max_nlocal() const
-    {
-        MIGRAPHX_ASSERT(blockDim.x > 0);
-        return blockDim.x;
```

Comment:
Dont delete this, just call `compute_max_local_size` here.

Reply:
That's what it's doing.